security/password_migration.rst

```diff
@@ -0,0 +1,212 @@
+
+Configure a new Encoder Using "migrate_from"
+--------------------------------------------
+
+When configuring a new encoder, you can specify a list of legacy encoders by
+using the ``migrate_from`` option:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/security.yaml
+        security:
+            # ...
+
+            encoders:
+                legacy:
+                    algorithm: sha256
+                    encode_as_base64: false
+                    iterations: 1
+
+                App\Entity\User:
+                    # the new encoder, along with its options
+                    algorithm: sodium
+                    migrate_from:
+                        - bcrypt # uses the "bcrypt" encoder with the default options
+                        - legacy # uses the "legacy" encoder configured above
+
+    .. code-block:: xml
+
+        <!-- config/packages/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:security="http://symfony.com/schema/dic/security"
+            xsi:schemaLocation="http://symfony.com/schema/dic/security
+                https://symfony.com/schema/dic/security/security-1.0.xsd">
+
+            <security:config>
+                <!-- ... -->
+
+                <security:encoder class="legacy"
+                    algorithm="sha256"
+                    encode-as-base64="false"
+                    iterations="1"
+                />
+
+                <!-- algorithm: the new encoder, along with its options -->
+                <security:encoder class="App\Entity\User"
+                    algorithm="sodium"
+                >
+                    <!-- uses the bcrypt encoder with the default options -->
+                    <security:migrate-from>bcrypt</security:migrate-from>
+
+                    <!-- uses the legacy encoder configured above -->
+                    <security:migrate-from>legacy</security:migrate-from>
+                </security:encoder>
+            </security:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        $container->loadFromExtension('security', [
+            // ...
+
+            'encoders' => [
+                'legacy' => [
+                    'algorithm' => 'sha256',
+                    'encode_as_base64' => false,
+                    'iterations' => 1,
+                ],
+
+                'App\Entity\User' => [
+                    // the new encoder, along with its options
+                    'algorithm' => 'sodium',
+                    'migrate_from' => [
+                        'bcrypt', // uses the "bcrypt" encoder with the default options
+                        'legacy', // uses the "legacy" encoder configured above
+                    ],
+                ],
+            ],
+        ]);
+
+.. tip::
```

Comment:
@chalasr please review this tip, I couldn't really follow the EncoderFactory/SecurityExtension dance that was doing this